front/src/common/ErrorBoundary.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { useRouteError } from 'react-router-dom';
+import { SerializedError } from '@reduxjs/toolkit';
+import PageNotFound from './PageNotFound';
+import { ApiError } from './api/emptyApi';
+
+export default function ErrorBoundary() {
+  const error = useRouteError() as ApiError | SerializedError;
+  if ((error as ApiError).status === 404) {
+    return <PageNotFound error={error} />;
+  }
+  return <PageNotFound />;
+}
```

Comment:
I think it's strange that if there is an error which is not a 404, we redirect the user to a "not found page".
Maybe we can still display the error or at least have a "An error occured / Something went wrong" message (probably more secure) instead of a "page not found message".

Reply:
Check with @Math-R , it will be customized in a future PR.